blockly-core/core/ui/block_space/pan_drag_handler.js

```diff
@@ -182,17 +182,21 @@ Blockly.PanDragHandler.prototype.onPanDragTargetMouseDown_ = function (e) {
     this.onTargetMouseDown_();
   }
 
-  var isClickDirectlyOnDragTarget = e.target && e.target === this.target_;
+  var clickIsOnTarget = e.target && e.target === this.target_;
 
   // Clicking on the flyout background clears the global selection
-  if (Blockly.selected && !Blockly.readOnly && isClickDirectlyOnDragTarget) {
+  if (Blockly.selected && !Blockly.readOnly && clickIsOnTarget) {
     Blockly.selected.unselect();
   }
 
+  var blockUnmovable = Blockly.selected && !Blockly.selected.isMovable();
```

Comment:
The one thing strikes me as potentially dangerous about this is that I reach back out into the global "selected block" state to determine whether we're likely clicking through a block.

Because of this, I manually tested various permutations of (A) while having selected an unmovable block: (B) panning the blockspace (modal/non-modal), scrollable flyouts, dragging blocks and dragging blocks out of the flyout. They result in behavior that acts as expected, though a lot of that may rely on de-select behavior elsewhere in the event chain.

Another approach would be walking up the `e.target` chain and determining whether we're looking at a (top-level?) immovable block. My sticking point with that approach is that I'm not sure whether there's a great inexpensive way to find a block object based on any of a number of its SVG DOM children.